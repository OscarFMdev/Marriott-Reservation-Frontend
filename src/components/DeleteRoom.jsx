import React from 'react';
import DeleteRoomForm from './forms/DeleteRoomForm';

const RenderDeleteRoom = () => (
  <div>
    <DeleteRoomForm />
  </div>
);

export default RenderDeleteRoom;
