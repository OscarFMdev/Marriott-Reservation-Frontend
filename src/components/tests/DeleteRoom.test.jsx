import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import DeleteRoom from '../DeleteRoom';
import store from '../../redux/configureStore';

describe('It displays the Delete Rooom page title', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Router>
          <DeleteRoom />
        </Router>
      </Provider>,
    );
  });
  test('Delete Room', () => {
    expect(screen.getByText('Delete Rooms')).toBeDefined();
  });
});

test('Delete Rooms renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <DeleteRoom />
      </Router>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
