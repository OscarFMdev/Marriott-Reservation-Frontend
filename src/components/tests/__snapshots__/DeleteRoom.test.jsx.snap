// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Delete Rooms renders correctly 1`] = `
<div>
  <main
    className="_container_ca1eb0"
  >
    <h1>
      Loading...
    </h1>
    <h1>
      Delete Rooms
    </h1>
    <table>
      <thead>
        <tr>
          <th>
            Image
          </th>
          <th>
            Name
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            colSpan="3"
          >
            You have no rooms
          </td>
        </tr>
      </tbody>
    </table>
  </main>
</div>
`;
